File: week 2/gcd/GCD.java
import java.util.*;

public class GCD {
  private static int gcd(int a, int b) {
    // int current_gcd = 1;
    // for(int d = 2; d <= a && d <= b; ++d) {
    // }

    // return current_gcd;
	if (b == 0)
		return a;
	return gcd(b, a%b);
  }

  public static void main(String args[]) {
    Scanner scanner = new Scanner(System.in);
    int a = scanner.nextInt();
    int b = scanner.nextInt();

    System.out.println(gcd(a, b));
  }
}

File: week 2/gcd/gcd.cpp
#include <iostream>

int gcd(int a, int b) {
  // write your code here
  // int current_gcd = 1;
  // for (int d = 2; d <= a && d <= b; d++) {
    // if (a % d == 0 && b % d == 0) {
      // if (d > current_gcd) {
        // current_gcd = d;
      // }
    // }
  // }
  // return current_gcd;
  	if (b == 0)
		return a;
	return gcd(b, a%b);
}

int main() {
  int a, b;
  std::cin >> a >> b;
  std::cout << gcd(a, b) << std::endl;
  return 0;
}
